specification/mission/Mission.Management/model.tsp

```diff
@@ -0,0 +1,527 @@
+  @path
+  @pattern("^[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]$")
+  @doc("The name of the Enclave Connection Resource")
+  name: string;
+}
+
+@added(Microsoft.Mission.Versions.v2024_06_01_preview)
+@doc("EnclaveEndpoint Model Resource")
+@parentResource(EnclaveResource)
+model EnclaveEndpointResource is TrackedResource<EnclaveEndpointProperties> {
+  @key("enclaveEndpointName")
+  @segment("enclaveEndpoints")
+  @path
+  @pattern("^[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]$")
+  @doc("The name of the Enclave Endpoint Resource")
+  name: string;
+}
+
+@added(Microsoft.Mission.Versions.v2024_06_01_preview)
+@doc("CommunityEndpoint Model Resource")
+@parentResource(CommunityResource)
+model CommunityEndpointResource
+  is TrackedResource<CommunityEndpointProperties> {
+  @key("communityEndpointName")
+  @segment("communityEndpoints")
+  @path
+  @pattern("^[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]$")
+  @doc("The name of the Community Endpoint Resource")
+  name: string;
+}
+
+@added(Microsoft.Mission.Versions.v2024_06_01_preview)
+@doc("Approval Model Resource")
+model ApprovalResource is ExtensionResource<ApprovalProperties> {
+  @key("approvalName")
+  @segment("approvals")
+  @path
+  @pattern("^[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]$")
+  @doc("The name of the approvals resource.")
+  name: string;
+}
+
+@doc("Workload Interface")
+@armResourceOperations
+interface Workload {
+  get is ArmResourceRead<WorkloadResource>;
+  @useFinalStateVia("azure-async-operation")
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<
+    WorkloadResource,
+    LroHeaders = Azure.Core.Foundations.RetryAfterHeader &
+      ArmAsyncOperationHeader
+  >;
+  update is ArmResourcePatchAsync<WorkloadResource, WorkloadProperties>;
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes" "For backward compatibility"
+  #suppress "deprecated" "Backwards compatibility"
+  delete is ArmResourceDeleteWithoutOkAsync<WorkloadResource>;
+  listByEnclaveResource is ArmResourceListByParent<WorkloadResource>;
+  listBySubscription is ArmListBySubscription<WorkloadResource>;
+}
+
+@doc("Endpoint Interface")
+@armResourceOperations
+interface Endpoint {
+  get is ArmResourceRead<EndpointResource>;
+  @useFinalStateVia("azure-async-operation")
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<
+    EndpointResource,
+    LroHeaders = Azure.Core.Foundations.RetryAfterHeader &
+      ArmAsyncOperationHeader
+  >;
+  update is ArmResourcePatchAsync<EndpointResource, EndpointProperties>;
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes" "For backward compatibility"
+  #suppress "deprecated" "Backwards compatibility"
+  delete is ArmResourceDeleteWithoutOkAsync<EndpointResource>;
+  listByEnclaveResource is ArmResourceListByParent<EndpointResource>;
+  listBySubscription is ArmListBySubscription<EndpointResource>;
+}
+
+@doc("Virtual Enclave Interface")
+@armResourceOperations
+interface VirtualEnclave {
```

Comment:
It is a little weird that all operations are defined in a tsp file named `model.tsp`. Personally I think it is better to put all operations in a tsp file with more specific tsp file (e.g. `routes.tsp`)

Reply:
Thanks for the suggestion, addressed in [a8d0e084](https://github.com/Azure/azure-rest-api-specs/pull/32359/commits/a8d0e084bb1d0864fd5f933fe0bb9027802e3542)